samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleRenderer
+import com.arcgismaps.mapping.symbology.Symbol
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SingleTapConfirmedEvent
+import com.arcgismaps.mapping.view.geometryeditor.GeometryEditor
+import com.arcgismaps.mapping.view.geometryeditor.ReticleVertexTool
+import com.arcgismaps.mapping.view.geometryeditor.SnapRuleBehavior
+import com.arcgismaps.mapping.view.geometryeditor.SnapRules
+import com.arcgismaps.mapping.view.geometryeditor.SnapSourceEnablingBehavior
+import com.arcgismaps.mapping.view.geometryeditor.SnapSourceSettings
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.arcgismaps.utilitynetworks.UtilityAssetType
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import com.esri.arcgismaps.sample.snapgeometryeditswithutilitynetworkrules.R
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+
+class SnapGeometryEditsWithUtilityNetworkRulesViewModel(application: Application) : AndroidViewModel(application) {
+
+    // Define the map view proxy and map with basemap and initial extent
+    val mapViewProxy = MapViewProxy()
+
+    val arcGISMap = ArcGISMap(basemapStyle = BasemapStyle.ArcGISStreetsNight).apply {
+        // Geodatabase layers are always full extent, however if using feature service layers, we
+        // must ensure that tiles use full resolution in order to snap to features
+        loadSettings.featureTilingMode = FeatureTilingMode.EnabledWithFullResolutionWhenSupported
+        initialViewpoint = Viewpoint(
+            center = Point(-9811055.156028448, 5131792.19502501, SpatialReference.webMercator()),
+            scale = 1e4
+        )
+    }
+
+    // Get the file path of the geodatabase file
+    private val provisionPath: String by lazy {
+        application.getExternalFilesDir(null)?.path.toString() + File.separator +
+                application.getString(R.string.snap_geometry_edits_with_utility_network_rules_app_name)
+    }
+    private val filePath = provisionPath + application.getString(R.string.naperville_geodatabase)
+
+    // Create the mobile map package
+    private val geodatabase = Geodatabase(filePath)
+
+    // Hold references to the subtype sublayers for the distribution and service pipe layers
+    private var distributionPipeLayer: SubtypeSublayer? = null
+    private var servicePipeLayer: SubtypeSublayer? = null
+    private var pipelineLayerName = application.getString(R.string.pipeline_layer_name)
+    private var distributionPipeName = application.getString(R.string.distribution_pipe_name)
+    private var servicePipeLayerName = application.getString(R.string.service_pipe_layer_name)
```

Comment:
Should use `val`s here:
```suggestion
    private val pipelineLayerName = application.getString(R.string.pipeline_layer_name)
    private val distributionPipeName = application.getString(R.string.distribution_pipe_name)
    private val servicePipeLayerName = application.getString(R.string.service_pipe_layer_name)
```